plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixMetadata.java

```diff
@@ -337,10 +340,19 @@ public ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorT
                 .collect(toImmutableList());
         PhoenixOutputTableHandle phoenixOutputTableHandle = (PhoenixOutputTableHandle) beginInsert(session, plainTable, ImmutableList.copyOf(columns), retryMode);
 
+        // The TupleDomain for building the conjuncts of the primary keys
+        ImmutableMap.Builder<ColumnHandle, Domain> primaryKeysDomainBuilder = ImmutableMap.builder();
+        // This value is to build the TupleDomain, but it won't affect the query field in result of the `DefaultQueryBuilder#prepareDeleteQuery`
+        Domain dummy = Domain.singleValue(BIGINT, 0L);
+        for (JdbcColumnHandle columnHandle : phoenixClient.getPrimaryKeyColumnHandles(session, plainTable)) {
+            primaryKeysDomainBuilder.put(columnHandle, dummy);
+        }
```

Comment:
this is the only thing that's a bit weird to me.

I don't understand why we say "it won't affect query".

The `DefaultQueryBuilder` can change in future. Either way it does use the tuple domain to create conjuncts by passing this to `toConjuncts` method.

Reply:
Thanks for the review.
> I don't understand why we say "it won't affect query".
> 
The domain does not participate in building the conjuncts as long as it is the singValue domain, this is what I want, only use the column and bind exception. It's a bit awkward to reuse the `DefaultQueryBuilder`, but currently the `DefaultQueryBuilder` not support function only accept the `column handles` to only get sql string.

> The `DefaultQueryBuilder` can change in future. Either way it does use the tuple domain to create conjuncts by passing this to `toConjuncts` method.

You are right, so after the merge got accept, the `DefaultQueryBuilder` needs to keep the correctness of the result that with the parameters: column & singDomain.
We need either implement a new query builder or provide/extract a new function to support the current situation to let the `QueryBuilder` more easily to maintain, I can have a try on it.:)

Reply:
Thanks for the explanation.

> provide/extract a new function to support the current situation to let the QueryBuilder more easily to maintain, I can have a try on it.:)

I think this makes sense to do.